### PR TITLE
Update docker.distribution.project for kubernetes

### PR DESCRIPTION
**Pull request by sajithaliyanage**

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/editor/DistProjectEditorPage.java [hunk 1]
```diff
@@ -725,15 +736,34 @@ public Action getExportAction() {
             exportAction = new Action("Create Docker Image",
                     ImageDescriptor.createFromImage(SWTResourceManager.getImage(this.getClass(), "/icons/car.png"))) {
                 public void run() {
-                    exportDockerImage();
+                	DockerHubAuth newConfiguration = null;
+                	if (getContainerType().equals(DockerProjectConstants.KUBERNETES_CONTAINER)) {
+                		newConfiguration = new DockerHubAuth();
+                		DockerHubLoginWizard wizard = new DockerHubLoginWizard(newConfiguration);
+                    	IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/editor/DistProjectEditorPage.java]: Fix formatting issues.

File: plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/editor/DockerHubLoginPage.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	private static final String SUB_DIALOG_TITLE = "Enter Docker Registry Credentials";
+	private static final String EMPTY_STRING = "";
+	private static final String EMAIL =  "Email:";
+	private static final String USERNAME = "Username:";
+	private static final String PASSWORD = "Password:";
+	
+	public static final Pattern VALID_EMAIL_ADDRESS_REGEX = 
+		    Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+
+    private String dockerHubUsername = EMPTY_STRING;
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/editor/DockerHubLoginPage.java]: Fix formatting issues.

File: plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/model/ContainerProjectFieldValidator.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                throw new FieldValidationException("Container name cannot contain invalid characters(^/ : ; * # $ ? \" <> + $)");
+            }
+        }
+    }
+
+	private boolean isValidArtifactName(String name) {
+        Pattern pattern = Pattern.compile("^[^/\\ \\\\:@%\\^+;,=\\[\\{\\]\\}*#\\$?\"<>|\\(\\)]+$");
+        Matcher matcher = pattern.matcher(name);
+        return matcher.matches();
+    }
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/model/ContainerProjectFieldValidator.java]: Fix formatting issues.

File: plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/editor/DistProjectEditorPage.java (hunk 1 above)
Comment (sajithaliyanage) [reply plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/editor/DistProjectEditorPage.java]: Done

File: plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/model/ContainerProjectFieldValidator.java (hunk 3 above)
Comment (sajithaliyanage) [reply plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/model/ContainerProjectFieldValidator.java]: Done

File: plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/editor/DockerHubLoginPage.java (hunk 2 above)
Comment (sajithaliyanage) [reply plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/editor/DockerHubLoginPage.java]: Done

File: plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/model/DockerModel.java [hunk 4]
```diff
@@ -36,7 +36,21 @@ public boolean setModelPropertyValue(String key, Object data) throws ObserverFai
             setDockerRemoteRepository(data.toString());
         } else if (key.equalsIgnoreCase(DockerProjectConstants.DOCKER_REMOTE_TAG_KEY)) {
             setDockerRemoteTag(data.toString());
-        }
+		} else if (key.equalsIgnoreCase(DockerProjectConstants.KUBE_CONTAINER_NAME)) {
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/model/DockerModel.java]: Fix formatting issues.

Comment (sajithaliyanage) [reply plugins/org.wso2.developerstudio.eclipse.docker.distribution/src/org/wso2/developerstudio/eclipse/docker/distribution/model/DockerModel.java]: Done

**Pull request merged**